### PR TITLE
feat: `FlashSort` implementation

### PR DESCRIPTION
**Pull request by alxkm**
FlashSort implementation.
Adding standard test.

This PR implements this feature request https://github.com/TheAlgorithms/Java/issues/5304

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/sorts/FlashSort.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+ *     <li>Initializes a classification array `L` to keep track of the number of elements in each class.</li>
+ *     <li>Computes a normalization constant `c1` to map elements into classes.</li>
+ *     <li>Classifies each element of the array into the corresponding bucket in the classification array.</li>
+ *     <li>Transforms the classification array to compute the starting indices of each bucket.</li>
+ *     <li>Permutes the elements of the array into sorted order based on the classification.</li>
+ *     <li>Uses insertion sort for the final arrangement to ensure complete sorting.</li>
+ * </ol>
+ */
+public class FlashSort implements SortAlgorithm {
+    private static final double CLASSIFICATION_RATIO = 0.45;
```
[the remark this answers is not in the document]
Comment (alxkm) [reply src/main/java/com/thealgorithms/sorts/FlashSort.java:25]: It depends but, for education purposes (which this repository exist), making `CLASSIFICATION_RATIO` a variable, can add flexibility to implementation, allowing users to experiment with different ratios for potentially better performance or customization according to their specific data sets. 

So thanks, will add this.

File: src/test/java/com/thealgorithms/sorts/FlashSortTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        FlashSort sorter = new FlashSort(ratio);
+        assertEquals(ratio, sorter.getClassificationRatio());
+    }
+
+    @ParameterizedTest
+    @ValueSource(doubles = {0, 1, -0.1, 1.1})
+    public void testCustomConstructorInvalidRatio(double ratio) {
+        assertThrows(IllegalArgumentException.class, () -> new FlashSort(ratio));
+    }
+}
```
Comment (vil02) [inline src/test/java/com/thealgorithms/sorts/FlashSortTest.java:35]: Do you see some way to run the _sorting tests_ for different ratios? Currently only the `0.45` is really tested.

Comment (alxkm) [reply src/test/java/com/thealgorithms/sorts/FlashSortTest.java]: Good suggestion. 

This issue was resolved using Reflection and DynamicTest. We introduced flexibility in adjusting the classificationRatio for each test case. After setting the ratio, we use Reflection to retrieve the list of test methods from the superclass. For each classification ratio, we then generate a List<DynamicTest>, creating dynamic tests tailored to each specific ratio.

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**